File: lib/toggles.dart
import 'package:flutter/material.dart';

class ToggleSwitch extends StatefulWidget {
  final String name; // New variable to be passed as an argument

  ToggleSwitch({Key? key, required this.name}) : super(key: key);

  @override
  _ToggleSwitchState createState() => _ToggleSwitchState();
}

class _ToggleSwitchState extends State<ToggleSwitch> {
  bool isSwitched = false;

  void _toggleSwitch(bool value) {
    setState(() {
      isSwitched = value;
    });
  }
  @override
  Widget build(BuildContext context) {
    return Container(
      child: Column(
        children: [
          Switch(
            value: isSwitched,
            onChanged: _toggleSwitch,
            activeTrackColor: Colors.lightGreenAccent,
            activeColor: Colors.green,
          ),
          Text(
            isSwitched ? 'Switch is ON' : 'Switch is OFF',
            style: TextStyle(
              fontSize: 18,
              fontWeight: FontWeight.bold,
            ),
          ),
        ],
      ),
    );
  }
}

File: composeApp/src/commonMain/kotlin/org/whatsinmyfood/project/components/toggles.kt
/*import 'package:flutter/material.dart';

class ToggleSwitchContainer extends StatefulWidget {
  final Widget child;

  const ToggleSwitchContainer({Key? key, required this.child})
      : super(key: key);

  @override
  _ToggleSwitchContainerState createState() => _ToggleSwitchContainerState();
}

class _ToggleSwitchContainerState extends State<ToggleSwitchContainer> {
  @override
  Widget build(BuildContext context) {
    return widget.child;
  }
}

class ToggleSwitch extends StatelessWidget {
  final String passedName;
  final bool isHighlighted; // Indicates if the switch should be highlighted
  final Function(bool) onChanged; // Callback to handle changes

  const ToggleSwitch({
    Key? key,
    required this.passedName,
    required this.isHighlighted,
    required this.onChanged,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      child: Row(
        children: [
          Switch(
            value: isHighlighted,
            onChanged: (value) {
              onChanged(value); // Call the passed callback function
            },
            activeTrackColor: Colors.lightGreenAccent,
            activeColor: Colors.green,
          ),
          Text(
            passedName,
            style: const TextStyle(
              fontSize: 18,
              fontWeight: FontWeight.bold,
            ),
          ),
        ],
      ),
    );
  }
}

 */